packages/Popover/src/Popover.tsx

```diff
@@ -0,0 +1,226 @@
+
+const rootPopoverStyle = css`
+  transition: transform 150ms ease-in-out, opacity 150ms ease-in-out;
+  position: absolute;
+  pointer-events: none;
+  opacity: 0;
+`;
+
+export enum Align {
+  Top = 'top',
+  Bottom = 'bottom',
+  Left = 'left',
+  Right = 'right',
+}
+
+export enum Justify {
+  Start = 'start',
+  Middle = 'middle',
+  End = 'end',
+}
+
+interface PopoverProps {
+  /**
+   * Content that will appear inside of the popover component.
+   */
+  children: ReactNode;
+
+  /**
+   * Determines the active state of the popover component
+   *
+   * default: `false`
+   */
+  active: boolean;
+
+  /**
+   * Class name applied to popover content container.
+   */
+  className?: string;
+
+  /**
+   * Determines the alignment of the popover content relative to the trigger element
+   *
+   * default: `bottom`
+   */
+  align: Align;
+
+  /**
+   * Determines the justification of the popover content relative to the trigger element
+   *
+   * default: `start`
+   */
+  justify: Justify;
+
+  /**
+   * A reference to the element against which the popover component will be positioned.
+   */
+  refEl?: RefObject<HTMLElement>;
+
+  /**
+   * Specifies that the popover content will appear portaled to the end of the DOM,
+   * rather than in the DOM tree.
+   *
+   * default: `true`
+   */
+  usePortal: boolean;
+
+  /**
+   * Specifies the amount of spacing (in pixels) between the trigger element and the Popover content.
+   *
+   * default: `10`
+   */
+  spacing: number;
+}
+
+/**
+ * # Popover
+ *
+ * React Component that handles positioning of content relative to another element.
+ *
+ * ```
+<button>
+  <Popover active={true}>Hello world!</Popover>
+</button>
+```
+ * ---
+ * @param props.children Content to appear inside of Popover container.
+ * @param props.active Boolean to describe whether or not Popover is active.
+ * @param props.className Classname applied to Popover container.
+ * @param props.align Alignment of Popover component relative to another element: `top`, `bottom`, `left`, `right`.
+ * @param props.justify Justification of Popover component relative to another element: `start`, `middle`, `end`.
+ * @param props.refEl Reference element that Popover component should be positioned against.
+ * @param props.usePortal Boolean to describe if content should be portaled to end of DOM, or appear in DOM tree.
+ */
+function Popover({
+  children,
+  active,
+  className,
+  usePortal,
+  spacing,
+  align,
+  justify,
+  refEl,
+  ...rest
+}: PopoverProps): ReactElement {
+  const placeholderRef: React.RefObject<HTMLDivElement> = useRef(null);
+  const contentRef: React.RefObject<HTMLDivElement> = useRef(null);
+
+  // We use this to force a re-render on window resize
+  const [windowUpdateVal, setWindowUpdateVal] = useState(true);
+
+  useEffect(() => {
+    const forceUpdate = debounce(
+      () => setWindowUpdateVal(!windowUpdateVal),
+      100,
+    );
+
+    window.addEventListener('resize', forceUpdate);
```

Comment:
How about...we extract this logic into a custom hook (located within this file) so if we need to reuse it we're ready to.

Reply:
Got it!